### DISCUSSION TITLE
revalidateOnFocus triggered when windows.confirm() popup

### DISCUSSION THREAD

**Discussion by loonfly** [vercel/swr#670] [General] [answered]
Hi,

SWR has been working fine with me until today i realize my server got unexpected request when I click a button that causes a confirmation popup and **revalidateOnFocus** enabled. Please refer images below.

![image](https://user-images.githubusercontent.com/2266959/94635757-fb411b80-0305-11eb-8bc3-126b506f58e0.png)

![image](https://user-images.githubusercontent.com/2266959/94635786-08f6a100-0306-11eb-9e7f-486984d97729.png)

And this is the delete product function.

```
const handleDeleteProduct = (e) => {
    e.preventDefault();
    if (
      typeof window !== "undefined" &&
      window.confirm(
        "Are you sure you want to delete the product? Once deleted it cannot be recovered."
      )
    ) {
      deleteProduct(product.public_id);
    }
  };
```
Again, the problem here is the hook re-render and make a server request whenever i click "OK" or "CANCEL" on the confirmation box. 

my guess is that, SWR hook detects user loses focus on the component, then when i click "OK" or "CANCEL" the component re-focus, that's why resend server request.

**If**
If this is unavoidable because of default browser behaviour, mind suggest me a better approach to confirmation before actual delete?

**NOTE**
1) revalidateOnMount tested no issue here. the issue is revalidateOnFocus. 
2) This is the first time in my life to create a issue in github, please forgive me if i miss out anything here. 
3) SWR is a great tool. thanks with love.

**Comment by sergiodxa** [comment, marked as answer]
As you mention, this is how browser works, they move the focus out of the tab to the popup and then back to the tab.

What you can do is to disable `revalidateOnFocus` if that is a huge problem. Another option is that you don't use browser built-in popups and instead build your own one, this way to tab will not lose focus and it will not trigger the revalidation.

**Reply by loonfly** [reply]
hmmm got it. didn't realize popup move off the focus. Thanks! :)
